main.go

```diff
@@ -155,6 +197,79 @@ func findFile(r *zip.ReadCloser, name string) *zip.File {
 	return nil
 }
 
+func listAttributes(path string) {
+	if strings.HasSuffix(path, ".apk") || strings.HasSuffix(path, ".aab") {
+		fmt.Println("Listing attributes for", path)
+		listAttributesInZip(path)
+	} else {
+		fmt.Println("Listing attributes for AndroidManifest.xml")
+		printManifestAttributes(path)
+	}
+}
+
+func listAttributesInZip(path string) {
+	manifest, err := ioutil.TempFile(tmpDir, "AndroidManifest.*.xml")
+	if err != nil {
+		log.Fatalln("Failed creating temp file:", err)
+	}
+	defer os.Remove(manifest.Name())
+
+	var manifestPath string
+	if strings.HasSuffix(path, ".apk") {
+		manifestPath = "AndroidManifest.xml"
+	} else { // .aab
+		manifestPath = "base/manifest/AndroidManifest.xml"
+	}
+
+	extractFromZip(path, manifestPath, manifest)
+	printManifestAttributes(manifest.Name())
+}
+
+func printManifestAttributes(path string) {
+	in, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.Fatalln("Error reading file:", err)
+	}
+
+	xmlNode := &XmlNode{}
+	if err := proto.Unmarshal(in, xmlNode); err != nil {
+		log.Fatalln("Failed to parse manifest:", err)
+	}
+
+	for _, node := range xmlNode.GetElement().GetChild() {
+		if elem, ok := node.GetNode().(*XmlNode_Element); ok {
+			element := elem.Element
+			if element.GetName() == "uses-sdk" {
```

Comment:
You already used the `if ... continue` pattern below and here it could also be used to reduce the amount of indentation. Same with the other if check a two lines below.

Reply:
I thought I updated this. WIll make sure it gets updated.

Reply:
Also this got lost.